lib/docs/src/pagination.stories.tsx

```diff
@@ -1,24 +1,72 @@
 import { Meta } from "@storybook/react";
 import { useCallback, useState } from "react";
 import { Pagination } from "../../core/src";
+import { Utils } from "./utils";
 
-export default {
+const meta: Meta = {
 	title: "Components/Pagination",
 	component: Pagination,
-} as Meta;
+};
+
+Utils.page.component(meta, { sticky: true, shots: [] });
+
+export default meta;
 
 export const Primary = (): JSX.Element => {
-	const [page, setPage_] = useState(1);
-	const setPage = useCallback((page): Promise<void> => {
+	const [page, setPage] = useState(1);
+	return <Pagination value={page} setValue={setPage} max={10} min={1} />;
+};
+
+export const Basic = Primary;
+
+Utils.desc(Basic)(`
+Paginations are [controlled][1] components. You should maintain a [state][2]
+for the current page, and pass the control to a Pagination via its \`value\`
+and \`setValue\` props. A Pagination helps users go to any page between its
+\`min\` and \`max\` props, inclusive.
+
+[1]: https://reactjs.org/docs/forms.html#controlled-components
+[2]: https://reactjs.org/docs/hooks-state.html
+`);
+
+export const Async = (): JSX.Element => {
+	const [page, setPage] = useState(1);
+	const setPageAsync = useCallback((page): Promise<void> => {
 		return new Promise((resolve) => {
-			setPage_(page);
+			setPage(page);
 			window.setTimeout(() => resolve(), 1000);
 		});
 	}, []);
+	return <Pagination value={page} setValue={setPageAsync} max={10} min={1} />;
+};
+
+Utils.desc(Async)(`
+Commonly, you may need to fetch some data when users navigate to a new
+page. In these cases, return a [\`Promise\`][1] in the \`setValue\` callback to
+have the Pagination displays a loading state while waiting for data.
 
+[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
+`);
+
+export const OneOff = (): JSX.Element => {
+	// Zero-based API
+	const TOTAL_PAGES = 9;
+	const [page, setPage] = useState(0);
+
+	// One-based UI
 	return (
-		<div>
-			<Pagination value={page} setValue={setPage} max={10} min={1} />
-		</div>
+		<Pagination
+			value={page + 1}
+			setValue={(page) => setPage(page - 1)}
+			max={TOTAL_PAGES + 1}
+			min={1}
+		/>
 	);
 };
+
+Utils.desc(OneOff)(`
+In most cases, Paginations should start at "1" to mimic real-life counting.
+However, your pagination back end may start at "0". In these cases, it's
+recommended to use zero-based counting for your state and most logic, and only
+transfer to one-based in the Pagination's props.
+`);
```

Comment:
@lqt93 I had some thinking about a default values for `min` and `max` and I think at the moment we won't have default values for them:

- For "max", it's obvious that there is no sensible default values. In all cases, the user must set a max value.
- For "min", it's not really clear, since a "1" would be suitable for many cases. However, I still think it's not safe to have "1" as default value since it would make using a zero-based counting hard to use and easy to have buggy behaviour.

Instead, for now, I added a "OneOff" section on the doc to highlight the use case where the counting of BE and the displaying of Pagination does not use the same counter. I think it's good enough for now. We will re-consider using "1" as default value for "min" when we have more use cases. It won't be a breaking change anyway.

Reply:
Great. I agree with you.